### PR TITLE
Improvements in the Atmosphere RPC implementation

### PR DESCRIPTION
**Pull request by Starzu**
* WebSocket works in both directions and properly waits for a reconnection
* if WS is not available, it works for the client->server communication with returning values from the server to the client (server pushes are not supported)

@ddworak - take a look, please.

### CODE REVIEW

File: rpc/frontend/src/main/scala/io/udash/rpc/internals/ServerConnector.scala [hunk 1]
```diff
@@ -27,38 +27,55 @@ abstract class AtmosphereServerConnector[RPCRequest](private val serverUrl: Stri
...
+        onError = (res: AtmosphereResponse) => ready(false),
+        onClose = (res: AtmosphereResponse) => ready(false),
+        onClientTimeout = (res: AtmosphereResponse) => ready(false),
+        onMessage = (res: AtmosphereResponse) => handleMessage(res.responseBody),
+        onMessagePublished = (req: AtmosphereRequest, res: AtmosphereResponse) => handleMessage(req.responseBody)
+      )
+      Atmosphere.subscribe(atmRequest)
+    } else {
+      isReady = true
+      val atmRequest: AtmosphereRequest = AtmosphereRequest(
```
Comment (ddworak) [inline rpc/frontend/src/main/scala/io/udash/rpc/internals/ServerConnector.scala]: Extract common settings between these two branches and reuse, this is hard to read and even harder to maintain.

File: rpc/backend/src/main/scala/io/udash/rpc/AtmosphereService.scala [hunk 2]
```diff
@@ -38,20 +42,27 @@ trait AtmosphereServiceConfig[ServerRPCType] {
...
   extends AtmosphereServletProcessor with LazyLogging {
 
-  override def init(config: AtmosphereConfig): Unit =
-    BroadcastManager.init(config.getBroadcasterFactory, config.metaBroadcaster())
+  private var brodcasterFactory: BroadcasterFactory = _
 
-  override def onRequest(resource: AtmosphereResource): Unit = {
-    config.initRpc(resource)
+  override def init(config: AtmosphereConfig): Unit = {
+    brodcasterFactory = config.getBroadcasterFactory
```
Comment (ddworak) [inline rpc/backend/src/main/scala/io/udash/rpc/AtmosphereService.scala]: Do we ever want to re-initialize this? Make it a `val` and init in constructor unless we do.

Comment (Starzu) [inline rpc/backend/src/main/scala/io/udash/rpc/AtmosphereService.scala]: This method is called from Atmosphere internals, so we cannot pass the `BroadcasterFactory` as constructor argument.

File: core/frontend/src/main/scala/io/udash/utils/Logger.scala [hunk 3]
```diff
@@ -30,7 +30,7 @@ private object ConsoleLogger extends Logger {
 }
 
 /**
-  * Provides `logger` reference to [[io.udash.utils.ConsoleLogger]].
+  * Provides `logger` reference to io.udash.utils.ConsoleLogger.
```
Comment (ddworak) [inline core/frontend/src/main/scala/io/udash/utils/Logger.scala]: What's the reason for removing the doc link?

Comment (Starzu) [inline core/frontend/src/main/scala/io/udash/utils/Logger.scala]: `Could not find any member to link for "io.udash.utils.ConsoleLogger"` - it's private object.

**Pull request merged**